fluss-common/src/main/java/com/alibaba/fluss/row/serializer/BinarySerializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.row.serializer;
+
+import com.alibaba.fluss.memory.InputView;
+import com.alibaba.fluss.memory.OutputView;
+
+import java.io.IOException;
+
+import static com.alibaba.fluss.utils.VarLengthIntUtils.decodeInt;
+import static com.alibaba.fluss.utils.VarLengthIntUtils.encodeInt;
+
+/** Type serializer for {@code byte[]}. */
+public final class BinarySerializer extends SerializerSingleton<byte[]> {
```

Comment:
Why introduce the set of serializer? Currentlly, `IndexRowWriter` and `ComapctedRowWriter` and `ArrowWriter` have different serialization format for each data type, therefore, we don't have a shared serializer for the types.